### ISSUE TITLE
[Question] When moving files, does Nemo compare the file sizes before deleting from source? [and observations and comparisons with Windows Explorer]

### ISSUE THREAD

**Issue by HT-7** [linuxmint/nemo#2600]
When moving files, one should usually copy-verify-delete, especially through MTP, which is a deprecated and unstable protocol. But moving has the convenience benefits that space will be freed up on the source device immediately and that the files deleted from the source device are only the transferred ones (except for bogus file managers such as [(1)](https://support.google.com/photos/thread/106201?hl=en) [(2)](https://www.reddit.com/r/Android/comments/jj4mwl/warning_possible_bug_in_builtin_file_manager/.compact)) and one does not have to carefully re-select the correct files as previously.

But to make it safer and more reliable, **if it is not like that already**, I suggest Nemo to verify the file size of each file after moving, and delete it from source only then.

# Observations
I have observed that the implementation of MTP in Windows Explorer frankly **stinks.** Nemo handles MTP much more stable than Windows Explorer, especially in directories with many files which tends to make Windows Explorer freeze up indefinitely.

One distinction is that Windows Explorer shows files in a directory **while** the list of files is loading, while Nemo only after the file list loading has finished. I suggest an option in Nemo to **optionally show** the files of a partially loaded file list. This has the benefit that if I just want to make space free on the source device, I don't have to wait for a list of thousands of photo/video files to finish loading. But this just as a side note. I might open a separate issue for this.

## File moving behaviour

When moving files **within** a MTP storage, Windows copies and then deletes the files, rather than just renaming them.

When disconnecting the MTP device while files are moved from the PC to MTP device, Windows Explorer might end up deleting an incompletely transfered source file. 

And when moving files from the MTP device to the computer, each file will not be deleted individually after their transfer has finished, but all selected files will be deleted at once **only after** the **entire** file transfer has finished, which has two disadvantages:

1. Space on the source device (e.g. mobile phone) is **only** freed up after the entire selection has been transfered, rather than the first file.
2. If the file transfer is interrupted, whether manually or not, the files from the source device are not deleted, which means no space has been cleared from the source device. Then, the user is forced to carefully re-select the **exact same** files as before, and not inadvertedly an untransferred one, which would be annoying and inconvenient.

If I am in a hurry and need to free up space immediately, I just select all files from a huge directory (such as `DCIM`), and start the file transfer I know will not finish in time. When I need to go, I just click _cancel_ and disconnect. With Windows Explorer, no space would have been freed up.

I am very thankful that Nemo uses the superior **individual** deletion method when moving files.

**Comment by Jeremy7701** [comment]
When moving files _within_ a file system, Linux just alters the inode (a sort of directory entry) - so no data is copied or deleted. It's extremely fast.

When moving files _between_ file systems, the only sensible possibility is to do a full data copy and then delete the original data. It makes sense for nemo to get a notification that a file copy has completed normally before removing the original. Note that writes/updates to media are automatically checked much more strictly than just a simple file size check.

Privacy issues aside, I wouldn't trust these free cloud services with any important data - my photos are stored on my PC and regularly backed up.

If you have many thousands of files, have you considered organising them into sub-folders?

**Comment by HT-7** [comment]
> When moving files _within_ a file system, Linux just alters the inode (a sort of directory entry) - so no data is copied or deleted. It's extremely fast.

Yes, that is on mass storage (one of my favourite Linux features because it allows moving/renaming without interfering with other software accessing the file). But not on MTP or FT


> When moving files _between_ file systems, the only sensible possibility is to do a full data copy and then delete the original data. It makes sense for nemo to get a notification that a file copy has completed normally before removing the original. Note that writes/updates to media are automatically checked much more strictly than just a simple file size check.


Also on MTP and FTP?


> Privacy issues aside, I wouldn't trust these free cloud services with any important data - my photos are stored on my PC and regularly backed up.

Which is why [I prefer memory cards in mobile phones](https://en.EverybodyWiki.com/Benefits_of_expandable_storage). But that Googke link I shared is about Android's shitty precluded file manager, which users are increasingly forced to use because Google increasingly usurps superior third-party software with new restrictions in their operating system (and wonder why people are reluctant to update). 
 
> If you have many thousands of files, have you considered organising them into sub-folders?

Yes, but on mobile phones, the possibilities are limited due to storage freedom in Android being incrementally compromised since 2014, breaking compatibility with existing third-party apps increasingly. ES File Explorer has a highly useful [range selection feature](https://en.Wikipedia.org/ES_File_Explorer), but sadly more recent ones such as *Amaze File Manager* (open source) lack it so far. This means that if I wanted to select the 1000 last files of a folder with 5000 files, for example, I would have to tap 1000 files individually. **No, thanks.** That is no fun.

**Comment by Jeremy7701** [comment]
This is not a discussion forum.
You may report bugs or request new features in **nemo**.
I suggest you raise your issues in an appropriate place.

**Comment by icarter09** [comment]
https://forums.linuxmint.com/
